src/util/sharing_node.h

```diff
@@ -12,340 +12,499 @@ Author: Daniel Poetzl
 #ifndef CPROVER_UTIL_SHARING_NODE_H
 #define CPROVER_UTIL_SHARING_NODE_H
 
-#include <list>
+#define DEBUG
```

Comment:
If this is debug code this shouldn't be enabled by default. Also, "DEBUG" is too generic a name to have it defined and then include a bunch of headers. If you're going to keep this, you should give it a less generic name and immediately undef it after the ifdef check.